src/script/env-script.tsx

```diff
@@ -0,0 +1,35 @@
+// XXX: Blocked by https://github.com/vercel/next.js/pull/58129
+// import { headers } from 'next/headers';
+import { type FC } from 'react';
+
+import { type ProcessEnv } from '../typings/process-env';
+
+type EnvScriptProps = {
+  env: ProcessEnv;
+};
+
+/**
+ * Sets the provided environment variables in the browser. If an nonce is
+ * available, it will be set on the script tag.
+ *
+ * Usage:
+ * ```ts
+ * <head>
+ *   <EnvScript env={{ NODE_ENV: 'test', API_URL: 'http://localhost:3000' }} />
+ * </head>
+ * ```
+ */
+export const EnvScript: FC<EnvScriptProps> = ({ env }) => {
```

Comment:
Would be nice to receive the `nonce` via props and mark it as optional, probably most of people won't have CSP enabled.

Reply:
Nice suggestion, I've also added the option to configure the nonce header key.